### PR TITLE
fix: allow tofu/terraform caches to stay local

### PR DESCRIPTION
**Pull request by JonZeolla**
# Contributor Comments

This removes the `.terraformrc` and updates the related documentation. By pre-configuring a plugin cache in our images, we make it so that `init`s result in `.terraform/providers/...` directories which symlink to `/home/easy_infra/.terraform.d/plugin-cache/...` which not be what's intended by our users.

## Pull Request Checklist

Thank you for submitting a contribution to our project!

In order to streamline the review of your contribution we ask that you review and comply with the below requirements:

- [X] Rebase your branch against the latest commit of the target branch.
- [X] If you are adding a dependency, please explain how it was chosen.
- [X] If manual testing is needed in order to validate the changes, provide a testing plan and the expected results.
- [X] If there is an issue associated with your Pull Request, link the issue to the PR.
- [X] Validate that documentation is accurate and aligned to any project updates or additions.

Don't forget our more detailed contribution guidelines [here](https://github.com/SeisoLLC/operations/blob/main/documents/software-guidelines.md#contributing-to-a-repository).

### CODE REVIEW

File: build/Dockerfile.opentofu [hunk 1]
```diff
@@ -27,8 +25,6 @@ RUN curl -L https://github.com/env0/terratag/releases/download/${TERRATAG_VERSIO
  && chown root: /usr/local/bin/terratag \
  && su easy_infra -c "git clone https://github.com/tofuutils/tofuenv.git /home/easy_infra/.tofuenv --depth 1 --branch ${TOFUENV_VERSION}" \
  && rm -rf /home/easy_infra/.tofuenv/.git \
- && su easy_infra -c "mkdir -p /home/easy_infra/.terraform.d/plugin-cache" \
```
Comment (JonZeolla) [inline build/Dockerfile.opentofu:30]:
```suggestion
 && su easy_infra -c "mkdir -p /home/easy_infra/.terraform.d/plugin-cache" \
 && mkdir -p /home/easy_infra/.terraform.d/plugin-cache \
```

File: build/Dockerfile.terraform [hunk 2]
```diff
@@ -27,8 +25,6 @@ RUN curl -L https://github.com/env0/terratag/releases/download/${TERRATAG_VERSIO
  && chown root: /usr/local/bin/terratag \
  && su easy_infra -c "git clone https://github.com/tfutils/tfenv.git /home/easy_infra/.tfenv --depth 1 --branch ${TFENV_VERSION}" \
  && rm -rf /home/easy_infra/.tfenv/.git \
- && su easy_infra -c "mkdir -p /home/easy_infra/.terraform.d/plugin-cache" \
```
Comment (JonZeolla) [inline build/Dockerfile.terraform:30]:
```suggestion
 && su easy_infra -c "mkdir -p /home/easy_infra/.terraform.d/plugin-cache" \
 && mkdir -p /home/easy_infra/.terraform.d/plugin-cache \
```

**Review by JonZeolla — COMMENTED** [review]
Whups

File: build/Dockerfile.opentofu [hunk 3]
```diff
@@ -27,8 +25,6 @@ RUN curl -L https://github.com/env0/terratag/releases/download/${TERRATAG_VERSIO
...
- && mkdir -p /home/easy_infra/.terraform.d/plugin-cache \
  && mkdir -p /home/easy_infra/.local/bin/ \
```
Comment (JonZeolla) [inline build/Dockerfile.opentofu:28]:
```suggestion
 && su easy_infra -c "mkdir -p /home/easy_infra/.terraform.d/plugin-cache" \
 && mkdir -p /home/easy_infra/.terraform.d/plugin-cache \
 && mkdir -p /home/easy_infra/.local/bin/ \
```

File: build/Dockerfile.terraform [hunk 4]
```diff
@@ -27,8 +25,6 @@ RUN curl -L https://github.com/env0/terratag/releases/download/${TERRATAG_VERSIO
...
- && mkdir -p /home/easy_infra/.terraform.d/plugin-cache \
  && mkdir -p /home/easy_infra/.local/bin/ \
```
Comment (JonZeolla) [inline build/Dockerfile.terraform:28]:
```suggestion

 && su easy_infra -c "mkdir -p /home/easy_infra/.terraform.d/plugin-cache" \
 && mkdir -p /home/easy_infra/.terraform.d/plugin-cache \
 && mkdir -p /home/easy_infra/.local/bin/ \
```

File: build/Dockerfile.terraform [hunk 5]
```diff
@@ -27,6 +25,7 @@ RUN curl -L https://github.com/env0/terratag/releases/download/${TERRATAG_VERSIO
  && chown root: /usr/local/bin/terratag \
  && su easy_infra -c "git clone https://github.com/tfutils/tfenv.git /home/easy_infra/.tfenv --depth 1 --branch ${TFENV_VERSION}" \
  && rm -rf /home/easy_infra/.tfenv/.git \
+
```
Comment (JonZeolla) [inline build/Dockerfile.terraform:28]:
```suggestion
```

**Review by ChiefHolland — APPROVED** [review]
LGTM

**Pull request merged**